dm/master/coordinator/coordinator.go

```diff
@@ -431,8 +431,14 @@ func (c *Coordinator) restartMysqlTask(w *Worker, cfg *config.MysqlConfig) bool
 		// Error means there is something wrong about network, set worker to close.
 		// remove sourceID from upstreams. So the source would be schedule in other worker.
 		delete(c.upstreams, cfg.SourceID)
+		delete(c.workerToConfigs, w.Address())
 		w.SetStatus(WorkerClosed)
 	}
 	c.mu.Unlock()
+	if w.State() == WorkerClosed {
+		ectx, cancel := context.WithTimeout(c.etcdCli.Ctx(), etcdTimeouit)
+		defer cancel()
+		c.etcdCli.Delete(ectx, common.UpstreamBoundWorkerKeyAdapter.Encode(w.Address()))
```

Comment:
Do we need to handle the returned `error` here?

Reply:
Maybe an Error log?  It must receive a success response as long as the etcd cluster is still alive.

Reply:
I think log an error is fine, it still can help us to debug.